imageio/v2.py

```diff
@@ -278,6 +280,9 @@ def get_reader(uri, format=None, mode="?", **kwargs):
 
     image_file = imopen(uri, "r" + mode, **imopen_args)
 
+    if pilmode is not None:
+        kwargs["mode"] = pilmode
+
```

Comment:
Nice thinking 💯 This gets us 90% there but will, unfortunately, prevent using `pilmode` with the old pillow plugins (e.g., `PNG-PIL`) that we are in the process of deprecating. These plugins expect `pilmode` not `mode` and if we add this piece of logic here things will break.

Could you move it to the `LegacyReader` class constructor located here:

https://github.com/imageio/imageio/blob/c93536ca403863db3270990911ff438b7cf1e047/imageio/v2.py#L110-L113

`pilmode` will be passed as a `kwarg` (assuming we remove the explicit mention if it in `get_reader`), so we can check for it's existence via `kwargs["pilmode"]`.

We should also only change this kwarg if users use the `PillowPlugin`. We will need a check/guard for this as well, otherwise we might create surprising behavior for other plugins. We want to avoid importing the plugin (it requires pillow to be installed which is not always true) so the check will have to be something like `type(self.instance).__name__ == "PillowPlugin"`.

Does this make sense, or should I add more detail?

Reply:
No, this makes sense, I am jsut not sure about this part:
```    
if isinstance(image_file, LegacyPlugin):
    return image_file.legacy_get_reader(**kwargs)
else:
    return LegacyReader(image_file, **kwargs)
```
Moving the handling of pilmode to the LegacyReader will only handle it for the `else` branch, should it not be handled by the legacy_get_reader too?

Reply:
Yes that's correct and intentional; we don't need to do anything for the `LegacyPlugin` branch.

`LegacyPlugin` is a wrapper for plugins written for our old plugin system. This includes our old pillow plugin, which is the original reason for having `pilmode`. It will do the right thing without intervention from our side.

Trivia: The reason we have `pilmode` in the first place is that the old plugin system uses `mode` as a kwarg to decide if it should read or write. `mode` is also used by pillow to indicate the colorspace, and also by other backends to indicate various "modes". Obviously this is not ideal. Our new plugin uses `io_mode` to decide reading or writing, which frees up the name `mode` and allows plugins to use the name as a kwarg. Hence, our new pillow plugin can use `mode` to indicate the colorspace to use, which is what you would expect from pillow (since this is how it uses `mode`).

Reply:
That makes sense, thanks for clarifying that!